pythonPackages.gcovr: Fix build

knedlsepp opened the pull request:
###### Motivation for this change
This fixes pythonPackages.gcovr. Please backport to 19.03.
ZHF-19.03: #56826



###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

knedlsepp committed:
pythonPackages.gcovr: Fix build

knedlsepp commented:
@GrahamcOfBorg build python2Packages.gcovr python3Packages.gcovr

worldofpeace reviewed (commented):
Is this even used as python module?

worldofpeace commented:
Could you also use the `https` url for the homepage?

knedlsepp committed:
pythonPackages.gcovr: Fixup homepage

knedlsepp committed:
gcovr: Add alias to pythonPackages.gcovr

Since gcovr is most likely to be used as a script instead of a library,
we provide an alias to python3Packages.gcovr. We still keep
python2/3Packages.gcovr in case somebody really wants to import it as a
library, which can not entirely be ruled out.

knedlsepp commented:
@worldofpeace Done. Also added an alias to use the package simply as `gcovr`.

worldofpeace commented on pkgs/top-level/all-packages.nix:
```diff
@@ -9916,6 +9916,8 @@ in
 
   gcab = callPackage ../development/libraries/gcab { };
 
+  gcovr = with pythonPackages; toPythonApplication gcovr;
```
```suggestion
  gcovr = with python3Packages; toPythonApplication gcovr;
```

worldofpeace reviewed (changes requested):
Didn't see any tests being ran so they should be disabled
```
doCheck = false;
```

Or change the source to github and try to enable them.

worldofpeace commented:
ping @knedlsepp

knedlsepp commented:
@worldofpeace Sorry for the late reply. Did fixup the suggestions that you made. I tried switching to the github source for the tests, finally could make it work by disabling hardening flags and using gcc5 + some manual patching. Since the tests highly depend on the gcc version I figured it wasn't really worth the maintenance effort on our side.

worldofpeace commented:
> @worldofpeace Sorry for the late reply. Did fixup the suggestions that you made. I tried switching to the github source for the tests, finally could make it work by disabling hardening flags and using gcc5 + some manual patching. Since the tests highly depend on the gcc version I figured it wasn't really worth the maintenance effort on our side.

Indeed that is true. We can wait for upstream for that.

Thanks for completing this :sparkles: 
I've looked over the diff and it looks good, everything builds too.

worldofpeace reviewed (approved):
see above comment

worldofpeace commented:
backported in
* c23891f49d318df72c22d924f29157888a914044
*  b50c2a82aaeaa73cc3866e724c6cce68e2b47875
* a0b7a7913c422a791996532b1a72aa5ddb0c3536

knedlsepp commented:
Thx!
